scripts/opensearch-build/release.jenkinsfile

```diff
@@ -0,0 +1,104 @@
+properties([
+  parameters([
+    string(
+        name: 'DistManifestUrl',
+        description: 'The url to distribution manifest for this release')])
+])
+
+lib = library(
+    identifier: 'jenkins@uni-poc',
+    retriever:modernSCM(
+        [$class: 'GitSCMSource',
+        remote: 'https://github.com/peternied/opensearch-build.git',
+        traits: [gitBranchDiscovery()]
+        ])
+    )
+
+Map config = [
+    cleanup: { -> echo "Totally unneed clean for demo purposes"}
+]
+
+standardReleasePipeline(config) {
+    def distManifest = new FakeDistManifest(fakeYmlFromUrl(params.DistManifestUrl))
+
+    def downloadPath = fakeDownloadArtifacts(distManifest)
+
+    fakeSignArtifacts(
+        artifactsPath: downloadPath,
+        signatureType: '.sig',
+        distributionPlatform: 'linux',
+    )
+    
+    fakePublishToMaven(
+        mavenArtifactsPath: "${downloadPath}/maven",
+        autoPublish: true
+    )
+
+    fakePublishToArtifactRepositry(
+        productName: distManifest.product,
+        version: distManifest.version,
+    )
+
+    fakeCopyDockerImage(
+        sourceImagePath: distManifest.artifacts.containerReference,
+        destinationImagePath: "opensearchproject/${distManifest.product}:${distManifest.version}",
+        destinationType: "docker",
+        destinationCredentialIdentifier: "opensearch-docker-prod"
+    )
+
+    fakeCopyDockerImage(
+        sourceImagePath: distManifest.artifacts.containerReference,
+        destinationImagePath: "opensearchproject/${distManifest.product}:${distManifest.version}",
+        destinationType: "ecr",
+        destinationCredentialIdentifier: "opensearch-ecr-prod"
+    )
+}
+
+/////////////////////////////////////////////////////////////////////////////////////
+// Fakes were written to make sure POC could be executed against jenkins environment
+
+class FakeDistManifest implements Serializable {
+    // Additional contents need to be added to the dist manifest for
+    // these parameters
+    class Artifacts implements Serializable {
+        String containerReference
+        Artifacts(Map data) {
+            this.containerReference = data?.containerReference
+        }
+    }
+
+    Artifacts artifacts
+    String product
+    String version
```

Comment:
I'm not sure having the version in the manifest is ideal. For Data Prepper, I'd like to keep it in only the Gradle properties and not have to modify it in two places.

Reply:
My thinking is that the last output of the data prepper build process would be the manifest, with its data provided from gradle, making this distribution a record of what was built and where to get those things.